source/extensions/common/wasm/wasm.cc

```diff
@@ -93,6 +95,7 @@ Wasm::Wasm(WasmHandleSharedPtr base_wasm_handle, Event::Dispatcher& dispatcher)
                      toAbslStringView(base_wasm_handle->wasm()->wasm_vm()->runtime())));
                }),
       scope_(getWasm(base_wasm_handle)->scope_),
+      custom_stat_namespace_(getWasm(base_wasm_handle)->custom_stat_namespace_),
```

Comment:
what keeps base_wasm_handle valid through the lifetime of this? If the answer is "nothing" then custom_stat-namespace_ would become invalid.

It would be pretty low-cost to just unconditionally create the pool and call pool_.add() for each instance, if you just don't want to think about it :) Maybe about the same cost as you pay in overhead for the absl::optional declaration around the pool.

Reply:
OK made the change to explicitly create pool and call add() for each instance!